symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedLoopV2Test.java

```diff
@@ -38,335 +38,424 @@
-        verify(datafeedApi, times(1)).readDatafeed("test-id", "1234", "1234", ackId);
-        verify(datafeedApi, times(1)).readDatafeed("recreate-df-id", "1234", "1234", ackId);
-        verify(datafeedApi, times(1)).createDatafeed("1234", "1234");
-        verify(datafeedApi, times(1)).deleteDatafeed("test-id", "1234", "1234");
-        verify(datafeedApiClient, times(1)).rotate();
-    }
-
-    @Test
-    void testStartServerErrorDeleteDatafeed() throws ApiException {
-        AckId ackId = datafeedService.getAckId();
-        when(datafeedApi.listDatafeed("1234", "1234")).thenReturn(Collections.singletonList(new V5Datafeed().id("test-id")));
-        when(datafeedApi.createDatafeed("1234", "1234")).thenReturn(new V5Datafeed().id("recreate-df-id"));
-        when(datafeedApi.readDatafeed("test-id", "1234", "1234", ackId)).thenThrow(new ApiException(400, "client-error"));
-        when(datafeedApi.deleteDatafeed("test-id", "1234", "1234")).thenThrow(new ApiException(502, "client-error"));
-
-        assertThrows(NestedRetryException.class, this.datafeedService::start);
-        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
-        verify(datafeedApi, times(1)).readDatafeed("test-id", "1234", "1234", ackId);
-        verify(datafeedApi, times(2)).deleteDatafeed("test-id", "1234", "1234");
-        verify(datafeedApiClient, times(3)).rotate();
-    }
-
-    @Test
-    void testStartAuthErrorDeleteDatafeed() throws ApiException, AuthUnauthorizedException {
-        AckId ackId = datafeedService.getAckId();
-        when(datafeedApi.listDatafeed("1234", "1234")).thenReturn(Collections.singletonList(new V5Datafeed().id("test-id")));
-        when(datafeedApi.createDatafeed("1234", "1234")).thenReturn(new V5Datafeed().id("recreate-df-id"));
-        when(datafeedApi.readDatafeed("test-id", "1234", "1234", ackId)).thenThrow(new ApiException(400, "client-error"));
-        when(datafeedApi.deleteDatafeed("test-id", "1234", "1234")).thenThrow(new ApiException(401, "client-error"));
-        doThrow(AuthUnauthorizedException.class).when(authSession).refresh();
-
-        assertThrows(NestedRetryException.class, this.datafeedService::start);
-        verify(datafeedApi, times(1)).listDatafeed("1234", "1234");
-        verify(datafeedApi, times(1)).readDatafeed("test-id", "1234", "1234", ackId);
-        verify(datafeedApi, times(1)).deleteDatafeed("test-id", "1234", "1234");
-        verify(datafeedApiClient, times(2)).rotate();
-    }
+          return false;
+        }
+      });
+      when(datafeedApi.readDatafeed("test-id", "1234", "1234", new AckId().ackId("ack-id")))
+          .thenReturn(new V5EventList().addEventsItem(
+              new V4Event().type(RealTimeEventType.MESSAGESENT.name()).payload(new V4Payload())).ackId("finished"));
+      return null;
+    });
+
+    // we want to make sure the addANewListener is started and blocking before unblocking the first listener
+    Thread.sleep(100);
```

Comment:
How long does it take to execute in the end?

Reply:
100ms + a few more ms for it to do the rest